crates/uv-types/src/traits.rs

```diff
@@ -121,6 +121,21 @@ pub trait BuildContext {
         build_kind: BuildKind,
         build_output: BuildOutput,
     ) -> impl Future<Output = Result<Self::SourceDistBuilder>> + 'a;
+
+    /// Build by calling directly into the uv build backend without PEP 517, if possible.
+    ///
+    /// Checks if the source tree uses uv as build backend. If not, it returns `Ok(None)`, otherwise
+    /// it builds and returns the name of the built file.
```

Comment:
What does it return in the case of an editable?

Reply:
At this stage, the editable is a wheel that is indistinguishable from a regular wheel, except it contains a `.pth` file instead of the actual sources (but we still unpack it the regular way) and we're not allowed to cache it.